packages/extension-ui/src/Popup/Contacts/AddContact.tsx

```diff
@@ -0,0 +1,376 @@
+// Copyright 2019-2021 @polkadot/extension-ui authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import type { ThemeProps } from '../../types';
+
+import _ from 'lodash';
+import queryString from 'query-string';
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import { RouteComponentProps } from 'react-router';
+import styled from 'styled-components';
+
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import { Contact, Identity } from '@polkadot/extension-base/background/types';
+import { ContactsStore } from '@polkadot/extension-base/stores';
+import { decodeAddress, encodeAddress } from '@polkadot/keyring';
+import { hexToString, hexToU8a, isHex } from '@polkadot/util';
+
+import { ActionBar, ActionContext, ActionText, Button, InputWithLabel } from '../../components';
+import useTranslation from '../../hooks/useTranslation';
+import { Header } from '../../partials';
+import chains from '../../util/chains';
+
+const bs58 = require('bs58');
+
+/**
+ * Get address prefix
+ * Polkadot: 0
+ * Kusama: 2
+ * Substrate: 42
+ * others: ...
+ * @param address
+ */
+function getAddressPrefix (address: string): number {
```

Comment:
These are all functions in util-crypto - we really don't want to duplicate these here at all. This is already existing.

Reply:
Yeah, @AmazingBeerBelly let's replace the functions with those in util-crypto

Reply:
This is what also extracts the prefix with validation - https://github.com/polkadot-js/common/blob/master/packages/util-crypto/src/address/checksum.ts#L6 (last field in the tuple)